### PR TITLE
Add IBM-Z recipe and other fixes

### PR DESCRIPTION
**Pull request by ramereth**
This was originally just to add an IBM-Z recipe but this turned into a bug fixing fest. Here's a summary of changes that have been made in this PR:

- Added ibmz_ci recipe which creates a new Jenkins CI server for IBM-z slaves
- Changed from using ``Chef::EncryptedDataBagItem.load`` to ``data_bag_item`` which is more portable
- Move all the Jenkins Groovy logic from recipes into individual libraries
- Add TLS support
- Upgraded docker plugins
- Added SGE plugin configuration for powerci
- Properly restart jenkins after any plugins are installed/updated
- Refactor powerci/ibmz_ci to not restart every plugin on install
- Manually install copy-to-slave plugin (which disappeared from the plugin repo)
- Other misc improvements on tests

### CODE REVIEW

**Comment by ramereth** [comment]
[POWER CI: s390x docker slaves](https://trello.com/c/DgtsJli9/597-power-ci-s390x-docker-slaves)

**Comment by ramereth** [comment]
@TravisWhitehead updated the README!

File: README.md [hunk 1]
```diff
@@ -119,6 +119,7 @@ These environment variables are required on the following suites:
 - bumpzone
 - cookbook-uploader
 - github-comment
+- ibmz-ci
```
Comment (doublej472) [inline README.md]: This suite also needs OAuth authentication, and should be mentioned in the section below

**Comment by ramereth** [comment]
!bump minor *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v1.10.0.  Have a nice day!

**Pull request merged**